python/tank/api.py

```diff
@@ -592,10 +594,38 @@ def preview_filesystem_structure(self, entity_type, entity_id, engine=None):
                                                       engine)
         return folders
 
+    def get_site_cache_root(self):
+        """
+        Returns the location of the site cache root for this Tank instance.
+        """
+        # Invoke the global method which is generic and can compute any site
+        # cache root.
+        global get_site_cache_root
+        return get_site_cache_root(self.shotgun.base_url)
 
 ##########################################################################################
 # module methods
 
+
+def get_site_cache_root(base_url):
+    """
+    Returns the location of the site cache root based on a site.
+    :param base_url: Site we need to compute the root path for.
+    :returns: An absolute path to the site cache root.
+    """
+    if sys.platform == "darwin":
+        root = os.path.expanduser("~/Library/Caches/Shotgun")
+    elif sys.platform == "win32":
+        root = os.path.join(os.environ["APPDATA"], "Shotgun")
+    elif sys.platform.startswith("linux"):
+        root = os.path.expanduser("~/.shotgun")
```

Comment:
Caches are already segmented out on OSX, but is it worth adding a "cache" directory or something else for the other operating systems so that there isn't a bunch of project_**id** directories at the same level of something like the Desktop's site config and other things we'll probably eventually put in this directory (given the zipped config proposals)?